pkg/osvscanner/osvscanner.go

```diff
@@ -357,29 +357,68 @@ func scanDebianDocker(r *output.Reporter, query *osv.BatchedQuery, dockerImageNa
 	return nil
 }
 
-// Filters response according to config, returns number of responses removed
-func filterResponse(r *output.Reporter, query osv.BatchedQuery, resp *osv.BatchedResponse, configManager *config.ConfigManager) int {
-	hiddenVulns := map[string]config.IgnoreEntry{}
-
-	for i, result := range resp.Results {
-		var filteredVulns []osv.MinimalVulnerability
-		configToUse := configManager.Get(r, query.Queries[i].Source.Path)
-		for _, vuln := range result.Vulns {
-			ignore, ignoreLine := configToUse.ShouldIgnore(vuln.ID)
-			if ignore {
-				hiddenVulns[vuln.ID] = ignoreLine
-			} else {
-				filteredVulns = append(filteredVulns, vuln)
+// Filters results according to config, preserving order. Returns total number of vulnerabilities removed.
+func filterResults(r *output.Reporter, results *models.VulnerabilityResults, configManager *config.ConfigManager) int {
+	removedCount := 0
+	newResults := []models.PackageSource{}
+	for _, pkgSrc := range results.Results {
+		configToUse := configManager.Get(r, pkgSrc.Source.Path)
+		newPackages := []models.PackageVulns{}
+		for _, pkgVulns := range pkgSrc.Packages {
+			newVulns := filterPackageVulns(r, pkgVulns, configToUse)
+			removedCount += len(pkgVulns.Vulnerabilities) - len(newVulns.Vulnerabilities)
+			if len(newVulns.Vulnerabilities) > 0 {
+				newPackages = append(newPackages, newVulns)
 			}
 		}
-		resp.Results[i].Vulns = filteredVulns
+		if len(newPackages) > 0 {
+			pkgSrc.Packages = newPackages
+			newResults = append(newResults, pkgSrc)
+		}
 	}
+	results.Results = newResults
+
+	return removedCount
+}
 
-	for id, ignoreLine := range hiddenVulns {
-		r.PrintText(fmt.Sprintf("%s has been filtered out because: %s\n", id, ignoreLine.Reason))
+// Filters package-grouped vulnerabilities according to config, preserving ordering. Returns filtered package vulnerabilities.
+func filterPackageVulns(r *output.Reporter, pkgVulns models.PackageVulns, configToUse config.Config) models.PackageVulns {
+	hiddenVulns := map[string]bool{}
+	// Iterate over groups first to remove all aliases of ignored vulnerabilities.
+	newGroups := []models.GroupInfo{}
+	for _, group := range pkgVulns.Groups {
+		ignore := false
+		for _, id := range group.IDs {
+			var ignoreLine config.IgnoreEntry
+			if ignore, ignoreLine = configToUse.ShouldIgnore(id); ignore {
+				for _, id := range group.IDs {
+					hiddenVulns[id] = true
+				}
+				// NB: This only prints the first reason encountered in all the aliases.
+				r.PrintText(fmt.Sprintf("%s has been filtered out because: %s\n", ignoreLine.ID, ignoreLine.Reason))
```

Comment:
Wondering if this should be something like
```
VULN-XXX and N aliases have been filtered out because: [reason]
```

Reply:
Looks like a good format to me.